### PR TITLE
Fixed SpaceToBatch and BatchToSpace for 3d case and add testcases

### PR DESCRIPTION
**Pull request by steve-y**
### Details:
 - *Fixed SpaceToBatch and BatchToSpace ops to use format order not rank*
 - *Added functional testcases*

### Tickets:
 - *111074*

### CODE REVIEW

File: src/core/src/op/space_to_batch.cpp [hunk 1]
```diff
@@ -102,7 +102,7 @@ bool ngraph::op::v1::SpaceToBatch::evaluate_space_to_batch(const HostTensorVecto
 
     auto data_shape = data->get_shape();
 
-    if (!(data->get_shape().size() == 4 || data->get_shape().size() == 5)) {
```
Comment (wilson-seok) [inline src/core/src/op/space_to_batch.cpp:105]: Can we touch source code in core module directly? I think you should bring reviewer from OpenVINO core team and get confirmation.

Comment (steve-y) [reply src/core/src/op/space_to_batch.cpp:105]: I invited @ilyachur as a reviewer. Thank you.

Comment (ilyachur) [reply src/core/src/op/space_to_batch.cpp:105]: Could you please add new tests for batch_to_space/space_to_batch to template plugin? To cover reference implementation: https://github.com/openvinotoolkit/openvino/tree/master/src/plugins/template/tests/functional/op_reference

Comment (steve-y) [reply src/core/src/op/space_to_batch.cpp:105]: Added functional tests for batch_to_space/space_to_batch to template plugin. Could you please review it?

File: src/core/src/op/space_to_batch.cpp [hunk 2]
```diff
@@ -102,7 +102,7 @@ bool ngraph::op::v1::SpaceToBatch::evaluate_space_to_batch(const HostTensorVecto
...
+    if (!(data->get_shape().size() == 3 || data->get_shape().size() == 4 || data->get_shape().size() == 5)) {
```
Comment (ilyachur) [inline src/core/src/op/space_to_batch.cpp:105]: @jane-intel Do we really need this check? Can we use it for other ranks?

File: src/plugins/intel_gpu/src/plugin/ops/space_to_batch.cpp [hunk 3]
```diff
@@ -31,7 +31,7 @@ static void CreateSpaceToBatchOp(Program& p, const std::shared_ptr<ngraph::op::v
 
         std::vector<int32_t> sizes = inConst->cast_vector<int32_t>();
         int32_t default_size = i == 1 ? 1 : 0;
-        for (size_t s = sizes.size(); s < rank; s++) {
+        for (size_t s = sizes.size(); s < format.order().size(); s++) {
```
Comment (vladimir-paramuzov) [inline src/plugins/intel_gpu/src/plugin/ops/space_to_batch.cpp:34]:
```suggestion
        for (size_t s = sizes.size(); s < format.dimension(); s++) {
```

File: src/plugins/intel_gpu/src/plugin/ops/batch_to_space.cpp [hunk 4]
```diff
@@ -31,7 +31,7 @@ static void CreateBatchToSpaceOp(Program& p, const std::shared_ptr<ngraph::op::v
 
         std::vector<int32_t> sizes = inConst->cast_vector<int32_t>();
         int32_t default_size = i == 1 ? 1 : 0;
-        for (size_t s = sizes.size(); s < rank; s++) {
+        for (size_t s = sizes.size(); s < format.order().size(); s++) {
```
Comment (vladimir-paramuzov) [inline src/plugins/intel_gpu/src/plugin/ops/batch_to_space.cpp:34]:
```suggestion
        for (size_t s = sizes.size(); s < format.dimension(); s++) {
```

File: src/plugins/intel_gpu/src/plugin/ops/space_to_batch.cpp (hunk 3 above)
Comment (steve-y) [reply src/plugins/intel_gpu/src/plugin/ops/space_to_batch.cpp]: Applied your comments, and thanks.

File: src/plugins/intel_gpu/src/plugin/ops/batch_to_space.cpp (hunk 4 above)
Comment (steve-y) [reply src/plugins/intel_gpu/src/plugin/ops/batch_to_space.cpp]: Applied your comments, and thanks.

**Review by ilyachur — APPROVED** [review]

File: src/core/src/op/space_to_batch.cpp (hunk 2 above)
Comment (jane-intel) [reply src/core/src/op/space_to_batch.cpp:105]: I don't think so. As you can see [here](https://docs.openvino.ai/2023.0/openvino_docs_ops_movement_SpaceToBatch_2.html) -- the input could be nD, n>=2

**Pull request merged**